network.go

```diff
@@ -2073,11 +2129,14 @@ func (n *network) createLoadBalancerSandbox() error {
 		}
 	}()
 
-	endpointName := n.name + "-endpoint"
+	endpointName := n.lbEndpointName()
 	epOptions := []EndpointOption{
 		CreateOptionIpam(n.loadBalancerIP, nil, nil, nil),
 		CreateOptionLoadBalancer(),
 	}
+	if n.hasLoadBalancerEndpoint() && !n.ingress {
+		epOptions = append(epOptions, CreateOptionAnonymous())
```

Comment:
I guess worth spending a comment saying that Anonymous is to avoid it to be resolvable from the DNS